### ISSUE TITLE
Standalone Input panic when calling WithWidth

### ISSUE THREAD

**Issue by rusq** [charmbracelet/huh#87]
Hey team, first of all thanks for the great library.  I'm planning to replace the [archived] survey with huh, so just playing around with different methods.

I believe that it expects that the "theme" member variable to be populated, but when running the sample code below, it is nil.

The panic happens in `field-input.go:262` on v0.2.3:
```go
	frameSize := i.theme.Blurred.Base.GetHorizontalFrameSize()
```

And it looks like it can still happen on master ([permalink](https://github.com/charmbracelet/huh/blob/4bd4657a36ac2db392a0fa2a878773660b5e1e5b/field_input.go#L261))

Now, let's do the formalities:

**Describe the bug**
```go
	var workspace string
	huh.NewInput().
		Title("Enter Slack workspace name").
		Value(&workspace).
		Description("Blah").
		WithWidth(40).
		Run()
	return workspace, nil
```

Causes
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x191a54a]

goroutine 1 [running]:
github.com/charmbracelet/huh.(*Input).WithWidth(0xc000328e00, 0x28)
	/Users/rustam/go/pkg/mod/github.com/charmbracelet/huh@v0.2.3/field_input.go:262 +0x2a
github.com/rusq/slackdump/v2/auth/auth_ui.(*Huh).RequestWorkspace(0x223c438, {0xc000374a00?, 0xc0007bf050?})
	/Users/rustam/wp/_github/slackdump/auth/auth_ui/huh.go:18 +0x165
github.com/rusq/slackdump/v2/auth.NewRODAuth({0x223ab38, 0xc0004a7200}, {0xc0007bf1b0, 0x1, 0xc0000061a0?})
	/Users/rustam/wp/_github/slackdump/auth/rod.go:53 +0x15d
github.com/rusq/slackdump/v2/internal/app.SlackCreds.AuthProvider({{0x0?, 0x1bf33a0?}, {0x0?, 0x1413ff2?}}, {0x223ab38, 0xc0004a7200}, {0x0, 0x0}, 0x0, 0x0)
	/Users/rustam/wp/_github/slackdump/internal/app/auth.go:81 +0x2f6
github.com/rusq/slackdump/v2/internal/app.InitProvider({0x223ab38?, 0xc0004a71d0?}, {0xc0000e6720, 0x26}, {0x0, 0x0}, {0x2237d60, 0xc0002e2d00}, 0x29cde90?, 0x0)
	/Users/rustam/wp/_github/slackdump/internal/app/auth.go:155 +0x319
main.run({_, _}, {{{0x1, 0x0}, {0xc0004a68a0}, {{0x222c888, 0x1}, {0x1d17490, 0x4}, {0x0, ...}}, ...}, ...})
	/Users/rustam/wp/_github/slackdump/cmd/slackdump/main.go:146 +0x28f
main.main()
	/Users/rustam/wp/_github/slackdump/cmd/slackdump/main.go:110 +0x546
exit status 2
```

**Expected behavior**
Don't panic

**Screenshots**

**Desktop (please complete the following information):**
 - Version: v0.2.3

**Additional context**

**Comment by vitor-mariano** [comment]
You're right. The problem is happening because the input is evaluating its size before theming. I tried to fix it by evaluating the width after theming and it solved the panic, but I found more existing bugs lol

I noticed that `WithWidth` has no effect when called with Input because form and group override its value by the current terminal width (even in standalone inputs because `Input.Run()` is just a wrapper for a form with a single group). This also happens to other fields than Input.

I have a solution in mind and can work on the fix if the authors (cc @maaslalani @meowgorithm) haven't yet.

If you need a temporary solution in this version, you can create a form and set the width on it.

``` go
huh.NewForm(
	huh.NewGroup(
		huh.NewInput().
			Title("Enter Slack workspace name").
			Value(&workspace).
			Description("Blah"),
	),
).WithWidth(40).Run()
```

**Comment by maaslalani** [comment]
Ah interesting, thanks for the detailed bug report @rusq!

@vitor-mariano if I understand your comment correctly this happens only with the single input `Run` shorthand due to a bug with how form widths / input widths are handled? In that case, we'd be more than delighted to accept a PR for the fix but also happy to fix it ourselves! As it seems it's an issue with how the theme/width is being applied.

**Comment by vitor-mariano** [comment]
Not really. Let me detail it more.

1. The problems happen regardless of the standalone use. I just emphasized that a standalone input works similarly to a grouped input.
2. I checked the problems with `WithWidth` happen to other field types.
3. The panic is only one bug related to `WithWidth`.

Fixing the panic issue, I noticed a second bug. The lib ignores the width of the fields. This happens in two scenarios:
1. When you also set `WithWidth` in `form`, it overrides the width of all its children.
    ``` go
    huh.NewForm(
	    huh.NewGroup(
		    huh.NewInput().
			    Title("Enter Slack workspace name").
			    Value(&workspace).
			    Description("Blah").
			    WithWidth(40),
	    ),
    ).WithWidth(60).Run()
    ```
    The resulting size will be 60 because `form` propagates `WithWidth` to all its children.
2. When you don't set `WithWidth` in `form` it propagates the window size to all its children, similarly to when you set `WithWidth` explicitly:
    https://github.com/charmbracelet/huh/blob/main/form.go#L325-L331 
    ``` go
    case tea.WindowSizeMsg:
	    if f.width > 0 {
		    break
	    }
	    for _, group := range f.groups {
		    group.WithWidth(msg.Width)
	    }
    ```

I haven't tested groups yet, but I'm pretty sure the same bug applies.

**Comment by vitor-mariano** [comment]
I think the solution is to have two properties in all field types:
1. `maxWidth` that holds the desired field width.
2. the existing `width` will be the current rendered width.

To handle these properties, we could have the following functions:
1. A new function `Resize(width int)` will set the current width without changing the `maxWidth`.
2. The existing `WithWidth(width int)` function will set the `maxWidth` and resize.

This way, the form and groups can control the current width of its fields respecting the maximum width set by the developer.

**Comment by maaslalani** [comment]
This is fixed by #92
